os/tools/mkverifyappsize.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+############################################################################
+#
+# Copyright 2024 Samsung Electronics All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+############################################################################
+
+from __future__ import print_function
+import os
+import sys
+import math
+import config_util as util
+
+############################################################################
+#
+# This script verifies that L2 pages assigned per app are sufficient enough or not.
+#
+# parameter information :
+#
+# argv[1] is file path of binary file.
+# argv[2] is app dynamic RAM size.
+#
+############################################################################
+
+file_path  = sys.argv[1]
+dynamic_ram_size = int(sys.argv[2])
+cfg_path = os.path.dirname(__file__) + '/../.config'
+
+# Temporary file to estimate application size
+APP_SIZE_FILE = 'app_size_file'
+
+textsize = 0
+datasize = 0
+bsssize = 0
+l2_pg_per_app = int(util.get_value_from_file(cfg_path, "CONFIG_NUM_L2_PER_APP=").replace('"','').replace('\n',''))
+l2_pg_need = 0
+os.system('size ' + file_path + ' > ' + APP_SIZE_FILE)
+app_data = open(APP_SIZE_FILE, 'rb')
+line1 = app_data.readline().split()
+line2 = app_data.readline().split()
+size = dict(zip(line1,line2))
+if util.check_config_existence(cfg_path, 'CONFIG_XIP_ELF') == True :
```

Comment:
Can you please add comment as to why we need to calculate it differently in case of xip elf ? (flash and ram address are different)

Reply:
I have added comment for xip elf case (calculating l2 page table). Please check.